### PR TITLE
Makes parsing of change list more robust

### LINKED ISSUE #11: Change Category parser doesn't validate matchdata
The change category parser doesn't validate if the match data is actually filled.

https://github.com/wooga/atlas-releaseNotesGenerator/blob/c2ae5121ee75b8d312486c729683cc75680f49d2/src/main/groovy/wooga/gradle/releaseNotesGenerator/utils/ReleaseNoteBody.groovy#L73-L88

### PR DESCRIPTION
**Pull request by doubleday**
## Description
Parsing of change list is a bit picky and currently throws for some malformed change lists. This does not improve the parsing but prevents throwing the parse exception. Changes are dropped.

## Changes
* ![FIX] don't throw on change list parse errors

resolves [#11](https://github.com/wooga/atlas-releaseNotesGenerator/issues/11)

[NEW]:https://atlas-resources.wooga.com/icons/icon_new.svg "New"
[ADD]:http://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:http://resources.atlas.wooga.com/icons/icon_improve.svg "IMPROVE"
[CHANGE]:http://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:http://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:http://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:http://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:http://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:http://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:http://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:http://resources.atlas.wooga.com/icons/icon_webGL.svg "Web:GL"

### CODE REVIEW

File: src/main/groovy/wooga/gradle/releaseNotesGenerator/utils/ReleaseNoteBody.groovy [hunk 1]
```diff
@@ -79,11 +79,13 @@ class ReleaseNoteBody {
...
-                String text = match[0][2].toString()
-                new ChangeNote(category, text)
-            }
+                def match = (it =~ /\!\[(.*?)\] (.+)/)
+                if (match) {
+                    String category = match[0][1].toString()
+                    String text = match[0][2].toString()
+                    new ChangeNote(category, text)
+                }
+            } - null
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/releaseNotesGenerator/utils/ReleaseNoteBody.groovy:88]: What is this?

Comment (doubleday) [reply src/main/groovy/wooga/gradle/releaseNotesGenerator/utils/ReleaseNoteBody.groovy:88]: What do you mean with „this“?

Comment (Larusso) [reply src/main/groovy/wooga/gradle/releaseNotesGenerator/utils/ReleaseNoteBody.groovy:88]: `- null`

Comment (doubleday) [reply src/main/groovy/wooga/gradle/releaseNotesGenerator/utils/ReleaseNoteBody.groovy:88]: Ah this filters the null values that the collect call produces when there is no match. List - null

Comment (Larusso) [reply src/main/groovy/wooga/gradle/releaseNotesGenerator/utils/ReleaseNoteBody.groovy:88]: Uuhhh that is very ‘groovy’.

**Pull request merged**